docs/post-mortems/0005-app-stopped-working-after-eks-upgrade.md

```diff
@@ -0,0 +1,157 @@
+Normal   Created                 29m   kubelet                  Created container booking-server
+Normal   Started                 29m   kubelet                  Started container booking-server
+```
+
+* 16:00 AWS' support contact offered to help debugging.
+
+* 16:19 Kjøremiljø met with AWS' contact. During this session, we realized that the SecurityGroupPolicy for the failing application was missing a SecurityGroup. Kjøremiljø, having fixed issues with this [before](https://trello.com/c/y8VyNwRy/427-dns-doesnt-work-for-pods-in-securitygroups), remembered that applications using databases need two security groups, but the failing application only had one.
+
+* 16:35 We tested setting the correct security group, and verified that the application worked.
+
+## Impact
+<!--
+Describe the consequences this had for the organization.
+
+Example:
+- Approximately 30.000 users downloaded and experienced an error using the app.
+- Users already logged in experienced some minor delay in the app
+-->
+
+Booking's backend in development stopped working between 15:15 and 16:35. 
+
+Booking's frontend returned a web page, but it wasn't working properly.
+
+## Cause(s)
+<!--
+Describe relevant factors that played a part in causing the event.
+
+Example:
+The error happened due to a combination of:
+- A spike of new users
+- A bug in the backend code for user creation
+-->
+
+The error happened due to a combination of using EKS 1.20 together with application configuration (Kubernetes manifests) that were outdated.
+
+## Solution
+<!--
+Describe how the problem was fixed.
+
+Example:
+- Manually scaled up number of backend instances letting regular traffic stay unaffected
+- Created a fix for the database schema
+-->
+
+We updated to the correct application configuration:
+* removed `dnsPolicy: true` from all three application's Deployment.
+* Added `terminationGracePeriodSeconds: 30` to the backend Deployment.
+* Added the correct cluster security group to the backend's SecurityGroupPolicy.
+
+## What went well
+<!--
+Describe what went well trying to handle the event.
+
+Example:
+- Alerting mechanisms worked brilliantly when errors started comming in
+- Deploying the database update was fast
+-->
+
+* We knew by memory what a correct application configuration was. This would also be possible to aquire by running `okctl apply application` with an existing Application manifest, or generating a new on with `okctl scaffold application`. 
+* We ran the EKS upgrade in the development environment before the production environment
+* In the AWS account where we did the upgrade, we only has "Basic" support level, which probably would have been to slow to use. Luckily, someone on Slack was available to tell how to upgrade it - though we ended up fixing it before this became necessary.  
```

Comment:
If you log into the AWS account with the read only role, the support level will be shown as Basic. However, in some accounts (not quite sure if all or just a select few), if you use the admin role the support level will be shown as Business

Reply:
Thanks for the tip. (I was using admin role, but didn't know about this.)